Src/DirView.cpp

```diff
@@ -3264,14 +3265,61 @@ void CDirView::OnUpdateItemRename(CCmdUI* pCmdUI)
  */
 void CDirView::OnHideFilenames()
 {
+	CDiffContext& ctxt = GetDiffContext();
+	int selection_index;
+	String hiddden_item_path;
+	
 	m_pList->SetRedraw(FALSE);	// Turn off updating (better performance)
 	DirItemIterator it;
+
 	while ((it = SelRevBegin()) != SelRevEnd())
 	{
 		DIFFITEM &di = *it;
+		selection_index = it.m_sel;
+		hiddden_item_path = di.getItemRelativePath();
 		SetItemViewFlag(di, ViewCustomFlags::HIDDEN, ViewCustomFlags::VISIBILITY);
-		DeleteItem(it.m_sel);
-		m_nHiddenItems++;
+		DeleteItem(selection_index);
+		ctxt.m_vCurrentlyHiddenItems.push_back(hiddden_item_path);
+	}
+	m_pList->SetRedraw(TRUE);	// Turn updating back on
+}
+
+/**
+ * @brief determine if an item-relative-path is contained in the list of items to hide
+ */
+bool CDirView::IsItemToHide(const String& currentItem, const std::vector<String>& ItemsToHide) const
+{
+	return std::find(ItemsToHide.begin(), ItemsToHide.end(), currentItem) != ItemsToHide.end();
+}
+
+/**
+ * @brief hides items specified in the .winmerge file
+ */
+
+void CDirView::HideItems(const std::vector<String>& ItemsToHide)
```

Comment:
When TreeMode is enabled, hiding a file in a subfolder and saving the item in a project file works correctly, but loading the project file did not hide the file.
This is because the HideItems function only handles the items currently visible in the list view.
I would appreciate it if you could implement it in a way similar to the ExpandAllSubdirs function of DirActions.cpp.

Reply:
I have tried expanding the folders while searching for the item to hide, but have not tested for reason that I mention in other comment